src/HttpHeader.cc

```diff
@@ -1726,6 +1726,20 @@ HttpHeader::removeHopByHopEntries()
     }
 }
 
+void
+HttpHeader::removeIrrelevantContentLength(const Http::StatusCode code)
+{
+    // RFC 7230 section 3.3.2: A server MUST NOT send a Content-Length
+    // header field in any response with a status code of 1xx (Informational)
+    // or 204 (No Content).
+    if ((code == Http::scNoContent) || (code >= Http::scContinue && code < Http::scOkay)) {
```

Comment:
This condition feels like a layering violation to me: To the extent possible, headers should not know about status code-related rules. Also, this condition makes the whole check a bit more expensive -- this method will usually be called in vain.

I suggest moving this method to HttpReply and inlining it there.

To simplify the caller, I would also add an inlined function or two to the `StatusCode.h`. Here is a `StatusCode.h` sketch w/o documentation:

```
inline bool Is1xx(const StatusCode sc) { return scContinue <= sc && sc < scOkay; }
inline bool ProhibitsContentLength(const StatusCode sc) { return sc == scNoContent || Is1xx(sc); }
```

Other HTTP and FTP code can benefit from Http::Is1xx(). For example:

```
$ git grep '100.*< 200\b' src
src/clients/FtpRelay.cc:570:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/clients/FtpRelay.cc:581:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/clients/FtpRelay.cc:592:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/clients/FtpRelay.cc:604:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/clients/FtpRelay.cc:683:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/clients/FtpRelay.cc:697:    if (100 <= ctrl.replycode && ctrl.replycode < 200)
src/http.cc:754:    if (newrep->sline.protocol == AnyP::PROTO_HTTP && newrep->sline.status() >= 100 && newrep->sline.status() < 200) {
```

You may have to change Is1xx() argument type to get the FTP code compile -- I have not checked.

Reply:
Done.